src/index.ts

```diff
@@ -13,49 +13,94 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 import * as extend from 'extend';
 import * as path from 'path';
-import {AuthenticationConfig, Common, ServiceConfig} from '../third_party/types/common-types';
+import * as request from 'request';
+
+import {AuthenticationConfig, Common, ServiceConfig, ServiceObject} from '../third_party/types/common-types';
+
 import {Config, defaultConfig} from './config';
 import {Profiler, ProfilerConfig} from './profiler';
 
 const common: Common = require('@google-cloud/common');
+const metadataAPI = 'http://metadata.google.internal/computeMetadata/v1/';
+
+// Returns a promise which will resolve to true if running on GCE and false
+// otherwise.
+// It is assumed a program is running on GCE if the metadata API can be
+// accessed.
+function onGCE(): Promise<boolean> {
+  return new Promise<boolean>((resolve) => {
+    request.get(
```

Comment:
Will it fail fast enough when running outside of GCE? Or do we need to set a short timeout?

Also, this function could probably later be made a part of some shared code.

Reply:
I've run this and it does return pretty quickly, but I'll specify a timeout of 2 seconds, since that seems to be the maximum time golang's version of this function could take. (https://github.com/GoogleCloudPlatform/google-cloud-go/blob/master/compute/metadata/metadata.go#L249-L251)